src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -52,7 +52,14 @@ object EntityClient {
 
   case class ImportPFB(workspaceNamespace: String, workspaceName: String, pfbRequest: PfbImportRequest, userInfo: UserInfo)
 
-  def props(requestContext: RequestContext, modelSchema: ModelSchema)(implicit executionContext: ExecutionContext): Props = Props(new EntityClient(requestContext, modelSchema))
+  def props(entityClientConstructor: (RequestContext, ModelSchema) => EntityClient, requestContext: RequestContext,
+            modelSchema: ModelSchema)(implicit executionContext: ExecutionContext): Props = {
+    Props(entityClientConstructor(requestContext, modelSchema))
+  }
+
+  def constructor(app: Application)(requestContext: RequestContext,
+                                    modelSchema: ModelSchema)(implicit executionContext: ExecutionContext) =
+    new EntityClient(requestContext, modelSchema, app.googleServicesDAO)
```

Comment:
all of the `EntityClient.constructor`-related changes in this PR are required so that unit tests can pass a different `Application`,  containing a mock google dao.